src/__test__/CardEntry.test.ts

```diff
@@ -44,146 +48,151 @@ jest.mock('react-native', () => {
   return mockReactNative;
 });
 
+const nativeEventEmitter = new EventEmitter();
+
 describe('Test CardEntry', () => {
   beforeEach(() => {
     jest.clearAllMocks();
   });
 
-  it('startCardEntryFlow works with cardEntryDidObtainCardDetails callback', async (done) => {
+  it('startCardEntryFlow works with cardEntryDidObtainCardDetails callback', async () => {
     expect.assertions(3);
     try {
-      const mockCardDetails = { nonce: 'fake_nonce' };
+      const mockCardDetails:CardDetails = { nonce: 'fake_nonce', card: {} };
       const onCardNonceRequestSuccess = jest.fn();
+      nativeEventEmitter.addListener('cardEntryDidObtainCardDetails', (cardDetails:CardDetails) => {
+        onCardNonceRequestSuccess(cardDetails);
+      });
+      if (SQIPCardEntry === undefined) { return; }
       await SQIPCardEntry.startCardEntryFlow(null, onCardNonceRequestSuccess, null);
       expect(NativeModules.RNSQIPCardEntry.startCardEntryFlow).toHaveBeenCalledTimes(1);
-      MockEventEmitter.listeners.cardEntryDidObtainCardDetails(mockCardDetails);
+      nativeEventEmitter.emit('cardEntryDidObtainCardDetails', mockCardDetails);
       expect(onCardNonceRequestSuccess).toHaveBeenCalledTimes(1);
       expect(onCardNonceRequestSuccess).toHaveBeenCalledWith(mockCardDetails);
-      done();
```

Comment:
same as other test file update, I don't understand why do we remove `done`?

Reply:
It is not needed and becomes obsolete in the typescript test case.